.pre-commit-config.yaml

```diff
@@ -1,5 +1,6 @@
 default_language_version:
   python: "3.8"
+default_stages: ["pre-push"]
```

Comment:
Why this change?

Reply:
This is the one that may be able to be left alone.  This sets pre-commit to run only on push instead of after every commit.  I find having the pre-commit process run after every local commit a burden.  It feels like the rules should only be applied when you are pushing something back into the main repo.

Reply:
That actually makes a lot of sense, I like it!

Reply:
@cofin I backed this change out because it caused us to only run a couple of the hooks in CI.

>  It feels like the rules should only be applied when you are pushing something back into the main repo.

Agree - this is what I do but have manually managed the process by not installing the hook at all and running `make lint` once when I'm ready.

Just to experiment I created this as `litestar/.git/hooks/pre-push` and it seems to work:

```bash
#!/bin/sh

pdm run --venv 3.8 make lint
```

...although I get warnings like:

> Inside an active virtualenv /home/peter/.local/share/pdm/venvs/litestar-dj-FOhMr-3.8, reusing it.
Set env var PDM_IGNORE_ACTIVE_VENV to ignore it.

in the output - so maybe specifying the venv isn't necessary.